app/forms/otp_verification_form.rb

```diff
@@ -0,0 +1,22 @@
+class OtpVerificationForm
+  def initialize(user, code)
+    @user = user
+    @code = code
+  end
+
+  def submit
+    @success = user.authenticate_direct_otp(code)
```

Comment:
What I could do is move `user.authenticate_direct_otp(code)` into a `valid_direct_otp_code?` method, to be consistent with the TotpVerificationForm and RecoveryCodeForm.